src/ansi-c/type2name.cpp

```diff
@@ -202,7 +211,9 @@ static std::string type2name(
       if(it!=components.begin())
         result+='|';
       result+=type2name(it->type(), ns, symbol_number);
-      result+="'"+it->get_string(ID_name)+"'";
+      irep_idt component_name = it->get_name();
+      CHECK_RETURN(component_name != ID_nil);
```

Comment:
`nil` would actually be a valid name! `component_name.empty()` would be the error you are looking for.

Reply:
done